sky/clouds/cloud.py

```diff
@@ -61,6 +61,24 @@ class Cloud:
     def regions(cls) -> List[Region]:
         raise NotImplementedError
 
+    @classmethod
+    def regions_with_offering(cls, instance_type: Optional[str],
+                              accelerators: Optional[Dict[str, int]],
+                              use_spot: bool, region: Optional[str],
+                              zone: Optional[str]) -> Set[Region]:
+        """Look up the regions with the offerings.
+
+        When region or zone is not None, the returned value will be limited to
+        the specified region/zone.
+
+        Returns:
+            A set of `Region`s that have the offerings for the specified
+            resources.
+            For each `Region` in the set, `region.zones` is the list of in that
+            region that has the resources.
```

Comment:
```suggestion
            A set of `Region`s that offer the specified resources.
            For each `Region` in the set, `region.zones` is the list of zones in that
            region that offer the resources.
```

Reply:
Are  L77-78 accurate for clouds that don't expose "zones" in provisioning, e.g., Azure?

Reply:
Fixed to
```
For each `Region` in the set, `region.zones` is the list of `Zone`s
which have the offerings. For the clouds that do not expose `Zone`s,
`region.zones` is an empty list.
```